src/System.Transactions/tests/NonMsdtcPromoterTests.cs

```diff
@@ -2374,71 +2017,43 @@ public static void TestCase_SetDistributedIdAtWrongTime()
                     }
-                    catch (TransactionException ex)
+                    catch (Exception ex)
                     {
+                        Assert.True(ex is TransactionException || (ex is TargetInvocationException && ex.InnerException is TransactionException));
                         Trace(string.Format("Caught expected exception {0}:{1}", ex.GetType().ToString(), ex.ToString()));
                     }
-                    catch (TargetInvocationException ex)
-                    {
-                        if (ex.InnerException is TransactionException)
-                        {
-                            Trace(string.Format("Caught expected exception {0}:{1}", ex.GetType().ToString(), ex.ToString()));
-                        }
-                        else
-                        {
-                            TestFailed(testCaseDescription, string.Format("Unexpected TargetInvocationException InnerException {0}:{1}", ex.InnerException.GetType().ToString(), ex.InnerException.Message));
-                        }
-                    }
 
                     Promote(testCaseDescription, NonMsdtcPromoterTests.PromotedToken1);
 
                     Trace("After Promotion");
                     try
                     {
                         SetDistributedTransactionId(dummyPSPE, Transaction.Current, guidToSet);
-                        TestFailed(testCaseDescription, failureDetail);
+                        bool shouldNotBeExecuted = true;
+                        Assert.False(shouldNotBeExecuted);
                     }
-                    catch (TransactionException ex)
+                    catch (Exception ex)
                     {
+                        Assert.True(ex is TransactionException || (ex is TargetInvocationException && ex.InnerException is TransactionException));
                         Trace(string.Format("Caught expected exception {0}:{1}", ex.GetType().ToString(), ex.ToString()));
                     }
-                    catch (TargetInvocationException ex)
-                    {
-                        if (ex.InnerException is TransactionException)
-                        {
-                            Trace(string.Format("Caught expected exception {0}:{1}", ex.GetType().ToString(), ex.ToString()));
-                        }
-                        else
-                        {
-                            TestFailed(testCaseDescription, string.Format("Unexpected TargetInvocationException InnerException {0}:{1}", ex.InnerException.GetType().ToString(), ex.InnerException.Message));
-                        }
-                    }
 
                     ts.Complete();
                 }
             }
             catch (Exception ex)
             {
-                TestFailed(testCaseDescription, string.Format("Caught unexpected exception {0}:{1}", ex.GetType().ToString(), ex.ToString()));
-                return;
+                Assert.Null(ex);
```

Comment:
Yes, we can just let it propagate. I wasn't sure if that was "the thing to do" or if driving the test failure with an Assert would have been better. I will make this change.


---
In reply to: [82718999](https://github.com/dotnet/corefx/pull/12533#discussion_r82718999) [](ancestors = 82718999)